scripts/lib/sign-macos.ts

```diff
@@ -249,15 +249,27 @@ async function *findFilesToSign(dir: string): AsyncIterable<string> {
 /**
  * Detect if the path of a plain file indicates that it's the bundle executable
  */
-function isBundleExecutable(fullPath: string): boolean {
+async function isBundleExecutable(fullPath: string): Promise<boolean> {
   const parts = fullPath.split(path.sep).reverse();
 
   if (parts.length >= 4) {
-    // Foo.app/Contents/MacOS/Foo - the check style here avoids spell checker.
-    if (fullPath.endsWith(`${ parts[0] }.app/Contents/MacOS/${ parts[0] }`)) {
-      return true;
+    // Anything.app/Contents/MacOS/executable - the check style here avoids spell checker.
+    if (fullPath.endsWith(`.app/Contents/MacOS/${ parts[0] }`)) {
+      // Check Anything.app/Contents/Info.plist for CFBundleExecutable
+      const infoPlist = path.sep + path.join(...parts.slice(2).reverse(), 'Info.plist');
+
+      try {
+        const { stdout } = await spawnFile('/usr/bin/defaults', ['read', infoPlist, 'CFBundleExecutable'], { stdio: 'pipe' });
+
+        return stdout.trimEnd() === parts[0];
+      } catch {
+        log.info({ infoPlist }, 'Failed to read Info.plist, assuming not the bundle executable.');
+
+        return false;
+      }
```

Comment:
I just generalized the code here to check for CFBundleExecutable instead of assuming that the app and executable names are the same (they are not for the `Rancher Desktop.app`, where the executable is `applet`).

But I suspect that signing `Rancher Desktop.app` will sign **all** the files in `Rancher Desktop.app/Contents/MacOS` because I got a signing error when I had the shell script in there.

I'm not sure if the redundant signing causes any harm; maybe we can just return `true` for any file in the `MacOS` directory?

```suggestion
      return true;
```

But I'll leave this for another day...